packages/core/src/error.ts

```diff
@@ -96,17 +171,32 @@ class Neo4jError extends Error {
       error instanceof Neo4jError &&
       error.retriable
   }
+
+  /**
+   * The json string representation of the diagnostic record.
+   * The goal of this method is provide a serialized object for human inspection.
+   *
+   * @type {string}
+   * @public
+   */
+  public get diagnosticRecordAsJsonString (): string {
+    return json.stringify(this.diagnosticRecord, { useCustomToString: true })
+  }
 }
 
 /**
  * Create a new error from a message and error code
  * @param message the error message
- * @param code the error code
+ * @param {Neo4jErrorCode} [code] the error code
+ * @param {Neo4jError} [cause]
+ * @param {String} [gqlStatus]
+ * @param {String} [gqlStatusDescription]
+ * @param {ErrorDiagnosticRecord} diagnosticRecord - the error message
  * @return {Neo4jError} an {@link Neo4jError}
  * @private
  */
-function newError (message: string, code?: Neo4jErrorCode, cause?: Error): Neo4jError {
-  return new Neo4jError(message, code ?? NOT_AVAILABLE, cause)
+function newError (message: string, code?: Neo4jErrorCode, cause?: Neo4jError, gqlStatus?: string, gqlStatusDescription?: string, diagnosticRecord?: ErrorDiagnosticRecord): Neo4jError {
+  return new Neo4jError(message, code ?? NOT_AVAILABLE, gqlStatus ?? '50N42', gqlStatusDescription ?? 'error: general processing exception - unexpected error. ' + message, diagnosticRecord, cause)
```

Comment:
we have some constants defined at `notification.ts`. It might worth to extract some gql common to a single file.